osu.Game/Screens/Edit/Editor.cs

```diff
@@ -224,7 +215,40 @@ protected override bool OnExiting(Screen next)
                 Beatmap.Value.Track.Tempo.Value = 1;
                 Beatmap.Value.Track.Start();
             }
+
             return base.OnExiting(next);
         }
+
+        private void exportBeatmap() => host.OpenFileExternally(Beatmap.Value.Save());
+
+        private void onModeChanged(EditorScreenMode mode)
+        {
+            currentScreen?.Exit();
+
+            switch (mode)
+            {
+                case EditorScreenMode.Compose:
+                    currentScreen = new ComposeScreen();
+                    break;
+                case EditorScreenMode.Design:
+                    currentScreen = new DesignScreen();
+                    break;
+                default:
+                    currentScreen = new EditorScreen();
+                    break;
+            }
+
+            LoadComponentAsync(currentScreen, screenContainer.Add);
+        }
+
+        private void seek(int direction)
+        {
+            double amount = GetContainingInputManager().CurrentState.Keyboard.ShiftPressed ? 2 : 1;
```

Comment:
Can you not just take an `UIEvent` parameter to this method? We're potentially calling GetContainingInputManager() many times per frame now for precision scroll.